src/cbmc/cbmc_parse_options.cpp

```diff
@@ -904,6 +905,13 @@ bool cbmc_parse_optionst::process_goto_program(
     status() << "Partial Inlining" << eom;
     goto_partial_inline(goto_functions, ns, ui_message_handler);
 
+    if(cmdline.isset("string-refine"))
+    {
+      status() << "Preprocessing for java Character library" << eom;
```

Comment:
I made some changes to the code which allows it to be called during java bytecode conversion.
So it can be moved there.  See the commits 5801fbf , 848983b and 4c1986a